auth/src/main/java/com/firebase/ui/auth/util/data/PrivacyDisclosureUtils.java

```diff
@@ -0,0 +1,90 @@
+package com.firebase.ui.auth.util.data;
+
+import android.content.Context;
+import android.support.annotation.RestrictTo;
+import android.support.annotation.StringRes;
+import android.widget.TextView;
+
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.data.model.FlowParameters;
+import com.firebase.ui.auth.util.ui.PreambleHandler;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class PrivacyDisclosureUtils {
+
+    private static final int NO_TOS_OR_PP = -1;
+    public static void setupTermsOfServiceAndPrivacyPolicyText(Context context,
+                                                               FlowParameters flowParameters,
+                                                               TextView termsText) {
+        PreambleHandler.setup(context,
+                flowParameters,
+                getGlobalTermsStringResource(flowParameters),
+                termsText);
+    }
+
+    public static void setupTermsOfServiceFooter(Context context,
+                                                 FlowParameters flowParameters,
+                                                 TextView footerText) {
+        PreambleHandler.setup(context,
+                flowParameters,
+                getGlobalTermsFooterStringResource(flowParameters),
+                footerText);
+    }
+
+    public static void setupTermsOfServiceAndPrivacyPolicySmsText(Context context,
```

Comment:
Ok looking at this method I'd recommend a few things for code clarity:

  * Drop the last parameter here
  * Drop the `if/else` method in here
  * Where we call this method, do:
 
```java
if (singleProvider) {
   PrivacyUtils.blah()
} else {
    termsText.setText(multipleMessage)
}
```

I think that's easier to read and more similar to how we use the non-SMS methods.

Reply:
I agree. Changed it.